### ISSUE TITLE
Snowflake: Parsing exception when file format when selecting from staged data file

### ISSUE THREAD

**Issue by wong-codaio** [sqlfluff/sqlfluff#3445]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

SQLFluff is not able to parse a SELECT from staged data file when `file_format` is present in the statement.

### Expected Behaviour

SQLFluff is able to parse SELECT staging data file with `file_format` present in the statement.

### Observed Behaviour

SQLFluff throws an unparsable section exception.

Error
```
==== parsing violations ====
L:   3 | P:  16 |  PRS | Line 3, Position 16: Found unparsable section: '(file_format =>
                       | myformat) t'
```

### How to reproduce

* Create a SQL file (`example.sql`) containing the following (It is a simplified version of [this query from Snowflake doc](https://docs.snowflake.com/en/user-guide/querying-stage.html#example-2-calling-functions-when-querying-a-staged-data-file)):
```sql
SELECT
  t.$1, t.$2
FROM @mystage1 (file_format => myformat) t
;
```

* Run SQL lint `parse` option, using Snowflake as a dialect
```
$ sqlfluff parse example.sql --dialect snowflake
```

### Dialect

Snowflake

### Version

Version 0.13.2
```
(.venv) ~/code/sqlfluff (main) $ sqlfluff --version
sqlfluff, version 0.13.2
```

### Configuration

Default, with Snowflake as the dialect.

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by tunetheweb** [comment]
Fix proposed in #3446

**Comment by wong-codaio** [comment]
Thank you @tunetheweb!!
